### PR TITLE
jit: completes numeric instructions.

### PR DESCRIPTION
**Pull request by mathetake**

### CODE REVIEW

File: wasm/jit/compiler.go [hunk 1]
```diff
@@ -53,6 +53,11 @@ type compiler interface {
 	compileFConvertFromI(o *wazeroir.OperationFConvertFromI) error
 	compileF32DemoteFromF64() error
 	compileF64PromoteFromF32() error
+	compileI32ReinterpretFromF32() error
+	compileI64ReinterpretFromF64() error
+	compileF32ReinterpretFromI32() error
+	compileF64ReinterpretFromI64() error
+	compiledExtend(o *wazeroir.OperationExtend) error
```
Comment (codefromthecrypt) [inline wasm/jit/compiler.go:60]:
```suggestion
	compileExtend(o *wazeroir.OperationExtend) error
```

File: wasm/jit/jit_amd64_test.go [hunk 2]
```diff
@@ -4000,6 +4000,120 @@ func TestAmd64Compiler_compileF64PromoteFromF32(t *testing.T) {
...
+					compiler.releaseAllRegistersToStack()
+					compiler.returnFunction()
+
+					// Generate and run the code under test.
+					code, _, err := compiler.generate()
+					require.NoError(t, err)
+					eng := newEngine()
+					jitcall(uintptr(unsafe.Pointer(&code[0])), uintptr(unsafe.Pointer(eng)), 0)
+
+					// Reinterpret must preserve the bit-pattern.
```
Comment (codefromthecrypt) [inline wasm/jit/jit_amd64_test.go]: thanks for this comment

**Review by codefromthecrypt — APPROVED** [review]
lol last commit obviated most of my comments

**Comment by mathetake** [comment]
img

**Comment by codefromthecrypt** [comment]
all your numbers are belong to us

**Pull request merged**